lib/active_model/serializer/adapter/json/fragment_cache.rb

```diff
@@ -1,16 +1,6 @@
-require 'active_model/serializer/adapter/fragment_cache'
-module ActiveModel
-  class Serializer
-    class Adapter
-      class Json < Adapter
-        class FragmentCache
+class ActiveModel::Serializer::Adapter::Json::FragmentCache
```

Comment:
Great discussion, I also agree the readability is something we need to crave, and indeed we can remove `inherited` hook, and use `register`.
This is a huge refactor / adding some new interfaces already, let's keep it for now, merge it, and work this out on the namespace issue, how do you feel about it?